### ISSUE TITLE
Suggested format for comment styling

### ISSUE THREAD

**Issue by screenbeard** [fiatjaf/classless#3]
I'm interested in writing a theme using classless, but I was hoping to use it with a site I have that already has comments I want to import and keep. I could shoehorn them in, but I'm wondering if there would be a "right" way of laying out comments in the Classless manner. Searching around I can't find a mention of it.

**Comment by fiatjaf** [comment]
Yes, this is complicated. I tried to come up with a standard for comments, but decided to not do it before discussing it with other people. We could come up with something now. The problem is that none of the themes already support comments, and if we hardcode it and say "we have comments like this" it will become harder for anyone to create themes.

On the other hand, we could make it "optional" or perhaps even an "extension", meaning that we can have themes that support and comments that don't support comments, but this also may increase the mental burden of using the project. Nonetheless, comments are almost essential to blogs everywhere, so I feel we should support this.

What do you think?

**Comment by fiatjaf** [comment]
In the past I've [experimented in my personal website](https://github.com/fiatjaf/classless/blob/e332d9f7fbd74b5a2a8adb6199e2a3f86210525c/themes/jeen/src/style.css#L181) with putting comments on `main > article > section ul`. A `section` inside the main `article` does not make sense for me to anything else besides comments, so I guess that's a good place. Also, by making comments a list of items, someone could expand it later to support nested comments, which makes sense.

**Comment by fiatjaf** [comment]
[Ursprung](https://github.com/onli/ursprung/blob/5d86cc376f131ecc854a9eacf166cf6aa80b194a/designs/default/page.erb) has adapted the default Classless template and included a direct `ol` inside the `article`, but is styling it with specific classes. The comment form there is just a `form` after the `article`.

**Comment by screenbeard** [comment]
I think the idea of an extension would be good. I've had a go at creating a classless theme, and implemented comments in the way you've indicated on your personal website. It works well.

I've got a few more questions about the way you've implemented single pages vs lists of posts. Would that be better in a separate issue, or would you prefer if I emailed you?

**Comment by fiatjaf** [comment]
A separate issue is probably the best place.
